slice-core-api/src/main/java/com/cognifide/slice/api/tag/SliceLookupTag.java

```diff
@@ -49,14 +49,17 @@ public void doTag() throws JspException {
 			}
 
 			final PageContext pageContext = (PageContext) getJspContext();
-			final Object model = SliceTagUtils.getFromCurrentPath(pageContext, type, appName);
+			final Class typeClass = SliceTagUtils.getClassFromType(pageContext, type);
+			final Object model = SliceTagUtils.getFromCurrentPath(pageContext, typeClass, appName);
 			pageContext.setAttribute(var, model, PageContext.PAGE_SCOPE);
+		} catch (ClassNotFoundException e) {
+			throw new JspTagException("Could not get class for " + type);
 		} finally {
 			clean();
 		}
 	}
 
-	public void setType(Class<?> type) {
+	public void setType(String type) {

```

Comment:
Have a look at [this documentation](http://docs.oracle.com/javaee/5/tutorial/doc/bnann.html)
**Setting Deferred Value Attributes and Deferred Method Attributes** section in particular.

It mentions that i can have __setAttr(Object obj) __ and then check instanceOf obj.

I can do a test run of this change, but firstly what are your views on this change ?
Will the major version of Slice needs to be incremented, if i update only the setter signature and not the actual attribute type ?